clientconn.go

```diff
@@ -196,7 +195,7 @@ func WithTimeout(d time.Duration) DialOption {
 }
 
 // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
-func WithDialer(f func(addr string, timeout time.Duration) (net.Conn, error)) DialOption {
+func WithDialer(f transport.DialerFn) DialOption {
```

Comment:
I meant this needs to be 
`func WithDialer(f func(string, time.Duration, <-chan struct{}) (net.Conn, error)) DialOption`

I understood this introduces a bit code duplication but the callers should not know existence of the transport package.